charts/camunda-platform/test/integration/integration_test.go

```diff
@@ -92,6 +92,68 @@ func (s *integrationTest) TestServicesEnd2End() {
 	s.tryToLoginToOptimize()
 }
 
+func (s *integrationTest) TestServicesEnd2EndShouldFailWithUpgrade() {
+	// given
+	options := &helm.Options{
+		KubectlOptions: s.kubeOptions,
+	}
+	if _, err := k8s.GetPodE(s.T(), s.kubeOptions, s.release+"-zeebe-0"); err != nil {
+		helm.Install(s.T(), options, s.chartPath, s.release)
+	}
+
+	// when
+	err := helm.UpgradeE(s.T(), options, s.chartPath, s.release)
+
+	// then
+	s.Require().NotNil(err)
+}
+
+func (s *integrationTest) TestServicesEnd2EndWithUpgrade() {
+	// given
+	options := &helm.Options{
+		KubectlOptions: s.kubeOptions,
+	}
+	if _, err := k8s.GetPodE(s.T(), s.kubeOptions, s.release+"-zeebe-0"); err != nil {
+		helm.Install(s.T(), options, s.chartPath, s.release)
+	}
+	tasklistSecret := s.getSecret("-tasklist-identity-secret", "tasklist-secret")
+	operateSecret := s.getSecret("-operate-identity-secret", "operate-secret")
+	optimizeSecret := s.getSecret("-optimize-identity-secret", "optimize-secret")
+	keycloakAdminPassword := s.getSecret("-keycloak", "admin-password")
+	keycloakManagementPassword := s.getSecret("-keycloak", "management-password")
+	postgresqlPassword := s.getSecret("-postgresql", "postgres-password")
+
+	// when
+	upgradeOptions := &helm.Options{
+		KubectlOptions: s.kubeOptions,
+		SetStrValues: map[string]string{
+			"global.identity.auth.tasklist.existingSecret": tasklistSecret,
+			"global.identity.auth.optimize.existingSecret": optimizeSecret,
+			"global.identity.auth.operate.existingSecret":  operateSecret,
+			"identity.keycloak.auth.adminPassword":  keycloakAdminPassword,
+			"identity.keycloak.auth.managementPassword":  keycloakManagementPassword,
+			"identity.keycloak.postgresql.auth.password":  postgresqlPassword,
+		},
+	}
+	helm.Upgrade(s.T(), upgradeOptions, s.chartPath, s.release)
```

Comment:
:x: shouldn't we assert that this doesn't return an error? otherwise, the upgrade might fail silently and all the integration checks later will succeed only because the initial install was successful.

Reply:
This is the difference between `helm.Upgrade` and `helm.UpgradeE`. The last one will return the error.